node-commons/src/main/java/org/apache/iotdb/commons/utils/ThriftCommonsSerDeUtils.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.commons.utils;
+
+import org.apache.iotdb.common.rpc.thrift.TConsensusGroupId;
+import org.apache.iotdb.common.rpc.thrift.TConsensusGroupType;
+import org.apache.iotdb.common.rpc.thrift.TDataNodeLocation;
+import org.apache.iotdb.common.rpc.thrift.TEndPoint;
+import org.apache.iotdb.common.rpc.thrift.TRegionReplicaSet;
+import org.apache.iotdb.common.rpc.thrift.TSeriesPartitionSlot;
+import org.apache.iotdb.common.rpc.thrift.TTimePartitionSlot;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Utils for serialize and deserialize all the data struct defined by thrift-commons */
+public class ThriftCommonsSerDeUtils {
```

Comment:
It is better to use thrift's native serialization deserialization tool, which is more extensible in terms of compatibility. Let me show you an example.

```
TByteBuffer byteBuffer = new TByteBuffer(ByteBuffer.allocate(1000));
TCompactProtocol protocol = new TCompactProtocol(byteBuffer);
TSStatus status = new TSStatus(200);
status.setMessage("error200");
TSStatus subStatus = new TSStatus(300);
status.setSubStatus(Collections.singletonList(subStatus));
subStatus.setMessage("error300");
subStatus.setSubStatus(Collections.singletonList(new TSStatus(400)));
status.write(protocol);

byteBuffer.getByteBuffer().flip();

TSStatus c = new TSStatus();
c.read(protocol);
System.out.println(c);
```

Reply:
I left a TODO and I will optimize it later~